src/Behat/Gherkin/Node/FeatureNode.php

```diff
@@ -185,6 +192,14 @@ public function getBackground()
         return $this->background;
     }
 
+    /**
+     * @return RuleNode[]
+     */
+    public function getRules(): array
```

Comment:
[BC] We need to consider how a consumer will know this method is available. Maybe we need an interface to hint it? I slightly prefer

```php
if ($featureNode instanceof FeatureRulesNode) {
```

vs

```php
if(method_exists($featureNode, 'getRules')){
```

Reply:
Well, if a consumer does not want to have to check for existence, the easiest way is to bump their min version of behat/gherkin.

To me, there is no need for an interface (which would not be allowed to be removed without a BC break, and there is no right way to trigger deprecations for instanceof checks relying on deprecated interfaces, only static analysis can do it)